src/CollAction/Helpers/ImageFileManager.cs

```diff
@@ -0,0 +1,98 @@
+﻿using CollAction.Data;
+using CollAction.Models;
+using Microsoft.AspNetCore.Http;
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+namespace CollAction.Helpers
+{
+    public class ImageFileManager
+    {
+        private IFormFile _formFile = null;
+        private ImageFile _imageFile = null;
+        private string _fileName = "";
+
+        public ApplicationDbContext Context { get; set; }
+
+        public string WebRoot { get; set; }
+
+        public string WebFolder { get; set; }
+        
+        private string FileName { get { return String.Format("{0}{1}", _fileName, Path.GetExtension(_formFile.FileName).ToLower()); } }
+        
+        private string WebPath { get { return Path.Combine(WebFolder, FileName); } }
+
+        public async Task<ImageFile> UploadFormFile(IFormFile formFile, string fileName)
+        {
+            if (formFile == null) { return null; }
+
+            _formFile = formFile;
+            _fileName = fileName;
```

Comment:
It's probably not a good idea to have state in class that's only valid for one method call. Unnecessary state can be a source of bugs, and can make code harder to follow. Also, the state still exists afterwards while the FileManager keeps existing.

Prehaps a helper function would be easier, combined with passing formFile and fileName by parameter?
private string GetWebPath(string fileName, string extension) ...